### PR TITLE
Add boundPath clientTls module.

### PR DESCRIPTION
**Pull request by adleong**
Fixes https://github.com/BuoyantIO/linkerd/issues/64

boundPath is a client tls module that allows you to determine the common name based on the destination bound path.

e.g.

```
             |  client:
             |    tls:
             |      kind: io.l5d.clientTls.boundPath
             |      caCertPath: ${certs.caCert.getPath}
             |      names:
             |      - prefix: "/io.l5d.fs/{host}"
             |        commonNamePattern: "{host}.buoyant.io"
```

HttpsIntegrationTest is also split up and refactored to reduce duplication.

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -155,6 +155,17 @@ provided.  It must be an object containing keys:
...
+  destination bound path.  This plugin supports the following options:
+    * *caCertPath* -- Optional.  Use the given CA cert for common name
+    validation.
+    * *names* -- Required.  A list of name matchers which each must contain:
+      * *prefix* -- A path prefix.  All destinations which match this prefix
+      will use this entry to determine the common name.  Wildcards and variable
+      capture are allowed (see: `io.buoyant.linkerd.util.PathMatcher`)
+      * *commonNamePattern* -- The common name to use for destinations matching
+      the above prefix.  Variables captured in the prefix may be used in this
+      string.
```
Comment (esbie) [inline docs/config.md]: At this point the bullet indentations are making documentation hard to read. (eg, here we are documenting commonNamePattern which is a required option for a list of names, which is a required option for a tls plugin, whose class name is specified as a required option for tls.) Let's make separate headered sections for each tls plugin

Comment (adleong) [inline docs/config.md]: Agreed.  Fixed.

**Comment by esbie** [comment]
looks great! tioli: it could also be potentially useful to show some sample tls configurations in the docs, as this is one of the trickier config options :star:

**Comment by adleong** [comment]
Add example and rebase :star:

**Comment by esbie** [comment]
woop woop :+1:

**Pull request merged**